lms/services/grouping.py

```diff
@@ -33,52 +33,52 @@ def generate_authority_provided_id(
             tool_consumer_instance_guid, parent.lms_id, type_.value, lms_id
         )
 
-    def upsert_with_parent(  # pylint: disable=too-many-arguments
+    def upsert_with_parent(
         self,
-        tool_consumer_instance_guid,
-        lms_id,
-        lms_name,
-        parent: Grouping,
+        grouping_dicts: List[dict],
         type_: Grouping.Type,
-        extra=None,
+        parent: Optional[Grouping],
```

Comment:
Keep type_ and parent as top level params. It means they are common to all grouping_dicts but that behavior we support now. 

This makes the grouping_dicts thinner which I think is a nice side effect.

Reply:
Is `parent` really optional? I think we always pass it